unit-tests/src/main/scala/tests/Suite.scala

```diff
@@ -5,16 +5,23 @@ import sbt.testing.{EventHandler, Logger, Status}
 import scala.collection.mutable
 import scala.reflect.ClassTag
 
+final case class AssertionFailed(msg: String) extends Exception(msg)
+
```

Comment:
We should keep this PR dedicate to the round only. Can you please rebase it to exclude the test infra changes?